### DISCUSSION TITLE
Crates Index Partitioning Algorithm

### DISCUSSION THREAD

**Discussion by shellscape** [rust-lang/crates.io#4681] [Q&A] [answered]
👋  Ahoy hoy

I'm working on a project where we've got a list of crates and we'd like to independently track updates to them. The crates API, while very useful, does have some stern warnings about not abusing, and that's something that while tinkering, I wouldn't want to fall afoul of. It occurred to me that the index itself (https://github.com/rust-lang/crates.io-index) would be ideal, as I could just use git locally to parse updates in any shape. Unfortunately, I can't find any documentation about the partitioning algorithm that crates uses for the index, and I've been unable to locate it in code.

Could the crates team document that algorithm, or alternatively could someone point me to code for that algorithm?

Thanks!

**Comment by carols10cents** [comment, marked as answer]
By "partitioning algorithm", do you mean the directory structure for the index? If so, documentation for that is here: https://doc.rust-lang.org/nightly/cargo/reference/registries.html#index-format

**Reply by shellscape** [reply]
Wonderful, thank you.
